Warn user's when running make build locally

## Problem

I was developing coder locally on my macOS (Intel) and ran into issues with the `make build` script. Specifically, this error:


```shell
--- Creating archive for linux amd64 (/Users/jp/dev/coder/coder/dist/coder_0.7.6-devel+4f3202d9_linux_amd64.tar.gz)
a ./LICENSE
a ./README.md
a ./coder
/Users/jp/dev/coder/coder/dist/coder_0.7.6-devel+4f3202d9_linux_amd64.tar.gz


ln: illegal option -- P
usage: ln [-Ffhinsv] source_file [link_name]
       ln [-Ffhinsv] source_file ... linkname_dir
       link source_file link_name
make: *** [Makefile:34: build] Error 1
```

I believe we use `ln -P` in one place

https://github.com/coder/coder/blob/9bdaec6a2113c4c318f419d674e213d1b116c1b7/scripts/package.sh#L71-L75

## Solution

I see two solutions to fixing this problem.

### Option 1

Only pass `-P` on Linux-based OS's. It seems it's not [needed for macOS](https://ss64.com/osx/ln.html).

### Option 2

In scripts, check for an environment variable like `CI=1` and hard exit script with warning if not set.

--- a/scripts/build_docker.sh
+++ b/scripts/build_docker.sh
@@ -90,8 +90,8 @@ fi
 # hardlinked from.
 cdroot
 temp_dir="$(TMPDIR="$(dirname "$input_file")" mktemp -d)"
-ln -P "$input_file" "$temp_dir/coder"
-ln -P Dockerfile "$temp_dir/"
+ln "$input_file" "$temp_dir/coder"
+ln Dockerfile "$temp_dir/"
 
 cd "$temp_dir"
 

--- a/scripts/package.sh
+++ b/scripts/package.sh
@@ -68,11 +68,11 @@ fi
 # be hardlinked from.
 cdroot
 temp_dir="$(TMPDIR="$(dirname "$input_file")" mktemp -d)"
-ln -P "$input_file" "$temp_dir/coder"
-ln -P "$(realpath coder.env)" "$temp_dir/"
-ln -P "$(realpath coder.service)" "$temp_dir/"
-ln -P "$(realpath preinstall.sh)" "$temp_dir/"
-ln -P "$(realpath scripts/nfpm.yaml)" "$temp_dir/"
+ln "$input_file" "$temp_dir/coder"
+ln "$(realpath coder.env)" "$temp_dir/"
+ln "$(realpath coder.service)" "$temp_dir/"
+ln "$(realpath preinstall.sh)" "$temp_dir/"
+ln "$(realpath scripts/nfpm.yaml)" "$temp_dir/"
 
 cd "$temp_dir"
 

